CustomBatteries/API/CustomBatteriesService.cs

```diff
@@ -0,0 +1,48 @@
+﻿namespace CustomBatteries.API
+{
+    using System.Collections.Generic;
+    using System.Reflection;
+    using Common;
+
+    /// <summary>
+    /// An API service class that handles requests for CustomBatteries from external mods.
+    /// </summary>
+    /// <seealso cref="ICustomBatteriesService" />
+    public class CustomBatteriesService : ICustomBatteriesService
+    {
+        /// <summary>
+        /// Gets the full collection of <see cref="TechType" />s for all batteries, both vanilla and modded.
+        /// </summary>
+        /// <value>
+        /// The full collection of battery <see cref="TechType" />s.
+        /// </value>
+        /// <seealso cref="BatteryCharger" />
+        public HashSet<TechType> AllBatteries => BatteryCharger.compatibleTech;
+
+        /// <summary>
+        /// Gets the full collection of <see cref="TechType" />s for all power cells, both vanilla and modded.
+        /// </summary>
+        /// <value>
+        /// The full collection of power cell <see cref="TechType" />s.
+        /// </value>
+        /// <seealso cref="PowerCellCharger" />
+        public HashSet<TechType> AllPowerCells => PowerCellCharger.compatibleTech;
```

Comment:
Replace the `AllBatteries` and `AllPowerCells` properties with methods that return a new collection rather than pointing to the real collection (for safety).